cli/run_util.go

```diff
@@ -707,6 +707,18 @@ func activateAndRunSteps(
 			stepInfoPtr.Step.SourceCodeURL = pointers.NewStringPtr(*mergedStep.SourceCodeURL)
 		}
 
+		if mergedStep.RunIf != nil {
+			stepInfoPtr.Step.RunIf = pointers.NewStringPtr(*mergedStep.RunIf)
+		}
+
+		if mergedStep.Timeout != nil {
+			stepInfoPtr.Step.Timeout = pointers.NewIntPtr(*mergedStep.Timeout)
+		}
+
+		if mergedStep.NoOutputTimeout != nil {
+			stepInfoPtr.Step.NoOutputTimeout = pointers.NewIntPtr(*mergedStep.NoOutputTimeout)
+		}
```

Comment:
> These are needed because they were not set on the stepInfoPtr previously?

Yes, if you search for `stepInfoPtr.Step.` in the [activateAndRunSteps](https://github.com/bitrise-io/bitrise/blob/CI-318/cli/run_util.go#L578) func, you can see that we only set the step properties that are later used for presentation (step boxes).

`stepInfoPtr` is passed to the `buildRunResultCollector.registerStepRunResults`, which uses it for creating the stepResults:
- https://github.com/bitrise-io/bitrise/blob/CI-318/cli/build_run_result_collector.go#L73
- https://github.com/bitrise-io/bitrise/blob/CI-318/cli/build_run_result_collector.go#LL92

Then the added fields are used for creating the status reason: https://github.com/bitrise-io/bitrise/blob/CI-318/cli/build_run_result_collector.go#L184